python/src/scipp/table_html/formatting_html.py

```diff
@@ -0,0 +1,396 @@
+    else:
+        return var.variance if single else var.variances
+
+
+def _short_data_repr_html_non_sparse(var, variances=False):
+    if hasattr(var, "data"):
+        data_repr = repr(retrieve(var.data, variances))
+    else:
+        data_repr = repr(retrieve(var, variances))
+    return data_repr
+
+
+def _short_data_repr_html_sparse(var, variances=False):
+    s = []
+    non_sparse_dim = var.dims[0]
+    for i in range(var.shape[0]):
+        s.append(repr(retrieve(var[non_sparse_dim, i], variances)))
+
+    return "\n".join(s)
+
+
+def short_data_repr_html(var, variances=False):
+    """Format "data" for DataArray and Variable."""
+    if var.dtype == sc.dtype.string:
+        data_repr = str(retrieve(var, variances, single=True))
+    else:
+        if var.sparse_dim is None:
+            data_repr = _short_data_repr_html_non_sparse(var, variances)
+        else:
+            data_repr = _short_data_repr_html_sparse(var, variances)
+    return escape(data_repr)
+
+
+def format_dims(dims, sizes, coords):
+    if not dims:
+        return ""
+
+    dim_css_map = {
+        dim: " class='xr-has-index'" if dim in coords else "" for dim in dims
+    }
+
+    dims_li = "".join(
+        f"<li><span{dim_css_map[dim]}>" f"{escape(str(dim))}</span>: "
+        f"{size if size != sc.Dimensions.Sparse else 'Sparse' }</li>"
+        for dim, size in zip(dims, sizes)
+    )
+
+    return f"<ul class='xr-dim-list'>{dims_li}</ul>"
+
+
+def summarize_attrs_simple(attrs):
+    attrs_dl = "".join(
+        f"<dt><span>{escape(name)} :</span></dt>" f"<dd>{values}</dd>"
+        for name, values in attrs
+    )
+
+    return f"<dl class='xr-attrs'>{attrs_dl}</dl>"
+
+
+def summarize_attrs(attrs):
+    attrs_li = "".join(
+        f"<li class='xr-var-item'>\
+            {summarize_variable(name, values, is_attr=True)}</li>"
+        for name, values in attrs
+    )
+    return f"<ul class='xr-var-list'>{attrs_li}</ul>"
+
+
+def _icon(icon_name):
+    # icon_name is defined in icon-svg-inline.html
+    return (
+        "<svg class='icon xr-{0}'>"
+        "<use xlink:href='#{0}'>"
+        "</use>"
+        "</svg>".format(icon_name)
+
+    )
+
+
+def summarize_coord(dim, var):
+    is_index = dim in var.dims
+    return summarize_variable(str(dim), var, is_index)
+
+
+def summarize_coords(coords):
+    vars_li = "".join(
+        f"<li class='xr-var-item'>{summarize_coord(dim, values)}</span></li>"
+        for dim, values in coords)
+
+    return f"<ul class='xr-var-list'>{vars_li}</ul>"
+
+
+def summarize_variable(name, var, is_index=False, is_attr=None):
+    """
+    :param name:
+    :param var:
+    :param is_index: If the variable is an index - used to bold
+                     coordinates that represent the indices of
+                     a dimension that the data contains
+
+    :param is_attr: If the variable is for an attribute, then this
+                    hides the show/hide attributes button.
```

Comment:
I think xarray stores attributes with variables, but for scipp's variables we cannot have attributes. `DataArray` has the attributes in our case. That is, I think, we should hide the button also for coords, labels, masks.